File: dofun/safe/deviceinfo/utils/NetWorkUtils.java
package com.dofun.safe.deviceinfo.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.telephony.TelephonyManager;
import java.lang.reflect.Method;

/* loaded from: classes6.dex */
public class NetWorkUtils {
    public static boolean isNetworkConnected(Context context) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService("connectivity");
        NetworkInfo networkInfo = manager.getActiveNetworkInfo();
        return networkInfo != null && networkInfo.isAvailable();
    }

    public static boolean isMobileEnabled(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService("connectivity");
        try {
            Class cmClass = Class.forName(cm.getClass().getName());
            Method method = cmClass.getDeclaredMethod("getMobileDataEnabled", new Class[0]);
            method.setAccessible(true);
            return ((Boolean) method.invoke(cm, new Object[0])).booleanValue();
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    public static boolean isWifi(Context context) {
        NetworkInfo networkInfo;
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService("connectivity");
        if (manager == null || (networkInfo = manager.getNetworkInfo(1)) == null || !networkInfo.isConnectedOrConnecting()) {
            return false;
        }
        return true;
    }

    public static String getNetWorkType(Context context) {
        if (!isNetworkConnected(context)) {
            return "NONE";
        }
        if (isWifi(context)) {
            return "WIFI";
        }
        TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService("phone");
        if (telephonyManager != null) {
            int networkType = telephonyManager.getNetworkType();
            switch (networkType) {
                case 1:
                    return "GPRS";
                case 2:
                    return "EDGE";
                case 3:
                    return "UMTS";
                case 4:
                    return "CDMA";
                case 5:
                    return "EVDO_0";
                case 6:
                    return "EVDO_A";
                case 7:
                    return "1xRTT";
                case 8:
                    return "HSDPA";
                case 9:
                    return "HSUPA";
                case 10:
                    return "HSPA";
                case 11:
                    return "IDEN";
                case 12:
                    return "EVDO_B";
                case 13:
                    return "LTE";
                case 14:
                    return "EHRPD";
                case 15:
                    return "HSPAP";
                case 20:
                    return "5G";
            }
        }
        return "NONE";
    }
}

File: app/src/main/java/com/dofun/safe/deviceinfo/utils/NetWorkUtils.kt
package com.dofun.safe.deviceinfo.utils

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.telephony.TelephonyManager

object NetWorkUtils {
    fun isNetworkConnected(context: Context): Boolean {
        val manager = context.getSystemService("connectivity") as ConnectivityManager
        val networkInfo: NetworkInfo? = manager.activeNetworkInfo
        return networkInfo != null && networkInfo.isAvailable
    }

    fun isMobileEnabled(context: Context): Boolean {
        val cm = context.getSystemService("connectivity") as ConnectivityManager
        return try {
            val cmClass = Class.forName(cm.javaClass.name)
            val method = cmClass.getDeclaredMethod("getMobileDataEnabled")
            method.isAccessible = true
            (method.invoke(cm) as Boolean)
        } catch (e: Exception) {
            e.printStackTrace()
            false
        }
    }

    fun isWifi(context: Context): Boolean {
        val manager = context.getSystemService("connectivity") as ConnectivityManager?
        val networkInfo = manager?.getNetworkInfo(1)
        return networkInfo != null && networkInfo.isConnectedOrConnecting
    }

    fun getNetWorkType(context: Context): String {
        if (!isNetworkConnected(context)) {
            return "NONE"
        }
        if (isWifi(context)) {
            return "WIFI"
        }
        val telephonyManager = context.getSystemService("phone") as TelephonyManager?
        if (telephonyManager != null) {
            when (telephonyManager.networkType) {
                1 -> return "GPRS"
                2 -> return "EDGE"
                3 -> return "UMTS"
                4 -> return "CDMA"
                5 -> return "EVDO_0"
                6 -> return "EVDO_A"
                7 -> return "1xRTT"
                8 -> return "HSDPA"
                9 -> return "HSUPA"
                10 -> return "HSPA"
                11 -> return "IDEN"
                12 -> return "EVDO_B"
                13 -> return "LTE"
                14 -> return "EHRPD"
                15 -> return "HSPAP"
                20 -> return "5G"
            }
        }
        return "NONE"
    }
}
